src/modules/household/components/elements/JoinHouseholdDialog.tsx

```diff
@@ -0,0 +1,264 @@
+    error: fetchError,
+  } = useGetEnrollmentWithHouseholdQuery({
+    variables: {
+      id: conflictingEnrollmentId,
+    },
+    onCompleted: (data) => {
+      if (data?.enrollment?.household) {
+        const household = data?.enrollment?.household;
+        const initiator = household?.householdClients.find(
+          (hc) => hc.enrollment.id === conflictingEnrollmentId
+        );
+
+        if (initiator) {
+          // Once the household loads, set the joining clients list to the initiating client
+          // (the one with the conflicting enrollment), plus if they are the HoH, all their household members.
+          if (
+            initiator.relationshipToHoH ===
+            RelationshipToHoH.SelfHeadOfHousehold
+          ) {
+            setJoiningClients(household.householdClients);
+          } else {
+            setJoiningClients([initiator]);
+          }
+        }
+      }
+    },
+  });
+
+  const donorHousehold = useMemo(
+    () => conflictingEnrollment?.household,
+    [conflictingEnrollment]
+  );
+
+  const receivingHoh = useMemo(
+    () => findHohOrRep(receivingHousehold.householdClients),
+    [receivingHousehold.householdClients]
+  );
+
+  const receivingHohName = useMemo(() => {
+    return clientBriefName(receivingHoh.client);
+  }, [receivingHoh]);
+
+  const missingRelationshipsCount = useMemo(
+    () =>
+      joiningClients.filter((jc) => !relationships[jc.enrollment.id]).length,
+    [joiningClients, relationships]
+  );
+
+  const missingRelationshipsProps = useMemo(() => {
+    return {
+      disabled: missingRelationshipsCount > 0,
+      disabledReason:
+        missingRelationshipsCount > 0
+          ? `Missing ${missingRelationshipsCount} required fields`
+          : undefined,
+    };
+  }, [missingRelationshipsCount]);
+
+  const [joinedHousehold, setJoinedHousehold] = useState<
+    HouseholdFieldsFragment | undefined
+  >(undefined);
+  const [remainingHousehold, setRemainingHousehold] = useState<
+    HouseholdFieldsFragment | undefined
+  >(undefined);
+  const [joinHousehold, { loading: joinLoading, error: joinError }] =
+    useJoinHouseholdsMutation({
+      onCompleted: (data) => {
+        if (data.joinHouseholds) {
+          setJoinedHousehold(data.joinHouseholds.receivingHousehold);
+          setRemainingHousehold(
+            data.joinHouseholds.donorHousehold || undefined
+          );
+          onComplete?.(
+            data.joinHouseholds.receivingHousehold,
+            data.joinHouseholds.donorHousehold
+          );
+        }
+      },
+    });
+
+  const onSubmit = useCallback(() => {
+    if (missingRelationshipsCount > 0) return; // This should never happen; the button will be disabled
+
+    joinHousehold({
+      variables: {
+        input: {
+          receivingHouseholdId: receivingHousehold.id,
+          joiningEnrollmentInputs: joiningClients.map((hc) => {
+            return {
+              // todo @martha - discuss, should we use enrollmentLockVersion?
```

Comment:
Let's discuss whether to use `enrollmentLockVersion` here. I think I would propose de-scoping it and maybe ticketing it for a future improvement.